### PR TITLE
use php from the user’s path

### PR DESCRIPTION
**Pull request by macbookandrew**
Possible fix for #20

### CODE REVIEW

**Comment by mojowill** [comment]
I don't think this will work as `which php` won't have access to the same path variable running inside Alfred.

https://www.alfredforum.com/topic/17062-using-php-on-monterey-macos-12/?do=findComment&comment=90246

**Comment by macbookandrew** [comment]
The dreaded answer: “it works on my machine” 😆 

For context: I’m still on Big Sur and on an Intel machine. To be completely honest, I’ve done very little workflow development with Alfred, so am not that familiar with scripting in that context.

**Comment by tillkruss** [comment]
`which php` doesn't work on my M1, do you have an alternative? I assume you're on an Intel?

**Comment by mojowill** [comment]
`which php` will only work from an interactive shell (zsh) and if you have PHP installed via Homebrew or similar. The bundled PHP from Apple no longer exists.

**Comment by macbookandrew** [comment]
I thought `which php` works even in non-interactive shells, but only if you have a php binary accessible in your `PATH` (regardless of whether it’s bundled from the OS or installed via homebrew).

Sounds like the only real solution is to detect whether php is at `/usr/bin/php`, `/usr/local/bin/php`, or `/opt/homebrew/bin/php` and run it from there.

Not the most elegant, but this may work: 3494192a4a061220dd219819f44e3dab59c9c82c

**Comment by tillkruss** [comment]
Nah, LGTM 👍

**Comment by driftingly** [comment]
I had to update `"$1"` to  `"{query}"` to get it working on my M1 Monterey.

File: info.plist [hunk 1]
```diff
@@ -79,7 +79,13 @@
...
 				<string>Searching docs for "{query}"...</string>
 				<key>script</key>
-				<string>/opt/homebrew/bin/php laravel.php "$1"</string>
+				<string>if [ -f "/opt/homebrew/bin/php" ]; then
+    /opt/homebrew/bin/php laravel.php "$1"
+elif [ -f "/usr/local/bin/php" ]; then
+    /usr/local/bin/php laravel.php "$1"
+elif [ -f "/usr/bin/php" ]; then
+    /usr/bin/php laravel.php "$1"
+fi</string>
```
Comment (driftingly) [inline info.plist]:
```suggestion
				<string>if [ -f "/opt/homebrew/bin/php" ]; then
    /opt/homebrew/bin/php laravel.php "{query}"
elif [ -f "/usr/local/bin/php" ]; then
    /usr/local/bin/php laravel.php "{query}"
elif [ -f "/usr/bin/php" ]; then
    /usr/bin/php laravel.php "{query}"
fi</string>
```

Comment (macbookandrew) [reply info.plist]: @driftingly interesting…that change totally breaks it for me on the default setting “with input as argv”:

Screen Shot 2021-11-08 at 13 39 40

If I change my workflow settings to “with input as {query}” then `{query}` works. However, that results in changing [line 84](https://github.com/tillkruss/alfred-laravel-docs/blob/60d9f4d69442073f4298e39fdcd592f99162693c/info.plist#L83-L84):

```diff
				<key>scriptargtype</key>
-				<integer>1</integer>
+				<integer>0</integer>
```

Does that still work for you?

Comment (driftingly) [reply info.plist]: Interesting, "with input as argv" works as expected with `"$1"`.  
Updating `scriptargtype` to 0 works with `"{query}"`.

Not sure why mine would have been different than yours, but I have been messing with the settings when it wasn't working.

Comment (macbookandrew) [reply info.plist]: @tillkruss any preference on `$1` vs. `$query`?

**Comment by tillkruss** [comment]
We might as well stick with `{query}`, so old installs stay the same.

**Comment by macbookandrew** [comment]
Updated MR

**Pull request merged**